src/seeders/20201124160449-users.js

```diff
@@ -0,0 +1,111 @@
+const SCOPES = {
+  SITE_ACCESS: 1,
+  ADMIN: 2,
+  READ_WRITE_REPORTS: 3,
+  READ_REPORTS: 4,
+  APPROVE_REPORTS: 5,
+};
+
+const {
+  ADMIN, READ_WRITE_REPORTS, READ_REPORTS, APPROVE_REPORTS,
+} = SCOPES;
+
+const permissions = [
+  {
+    userId: 1,
+    scopeId: ADMIN,
+    regionId: 14,
+  },
+  {
+    userId: 3,
+    regionId: 2,
+    scopeId: READ_REPORTS,
+  },
+  {
+    userId: 3,
+    regionId: 2,
+    scopeId: READ_WRITE_REPORTS,
+  },
+  {
+    userId: 3,
+    regionId: 3,
+    scopeId: READ_REPORTS,
+  },
+  {
+    userId: 3,
+    regionId: 3,
+    scopeId: READ_WRITE_REPORTS,
+  },
+  {
+    userId: 4,
+    regionId: 14,
+    scopeId: ADMIN,
+  },
+  {
+    userId: 4,
+    regionId: 4,
+    scopeId: READ_WRITE_REPORTS,
+  },
+  {
+    userId: 4,
+    regionId: 4,
+    scopeId: APPROVE_REPORTS,
+  },
+];
+
+const users = [
+  {
+    id: 1,
```

Comment:
It seems weird to me to set the pks since they're an auto-incrementing id. It makes it so that you'd have to dump the db and reset ids on the table before running seed. Is this necessary?

Reply:
yeah we can drop the explicit id for most users but will need to keep an id for at least one user so we have a known id when `SKIP_AUTH` is true.

Reply:
We could also just use a well known name and remove all ids

Reply:
Instead of using a well known name, can we look up the user by `hsesUserId` or `email` since those are both unique?

Reply:
Wouldn't we run into the same problem using a unique field that we would if we used `id`, you'd have to dump the db and reset the unique field before seeding?

After thinking about this a bit we may want to keep using explicit IDs for seeds. In the future if we want to add additional seeders (for activity reports for example) we will need to tie those records to a user. Seems easier to have a few well known user ids we can use to tie everything together. Or we can create users in each new seed file? Might end up with a lot of users but probably doesn't matter that much.

Another option would be to use something like [Factory girl](https://github.com/simonexmachina/factory-girl) to handle all the dynamic associations for us. If we went that route I imagine we'd end up with a single seeder file that handles creating all our testing data. I've used this approach in the past with some success, granted it was with rails (using [Factory bot](https://github.com/thoughtbot/factory_bot)). Another advantage of using a factory library is it would probably speed up testing on the backend.

I'm not really sold on any approach right now, interested to hear other's thoughts.

Reply:
I am fine with either approach. Seems a bit easier/clearer to use explicit ids. The testing speed is a consideration, though. If you think the benefits would be substantial we could go that route.